locals.tf

```diff
@@ -66,6 +66,21 @@ locals {
     ] if gcr.repositories != null
   ])
 
+  # create gar_repositories list and sets default values for optional fields.
+  gar_repositories = [
+    for gar in var.gar_repositories : {
+      repo           = "${gar.region}-docker.pkg.dev/${gar.project_id != null ? gar.project_id : local.google_project_id}/${gar.name}"
+      grace          = gar.parameters != null ? (gar.parameters.grace != null ? gar.parameters.grace : "0") : "0"
+      keep           = gar.parameters != null ? (gar.parameters.keep != null ? gar.parameters.keep : "0") : "0"
+      tag_filter     = gar.parameters != null ? (gar.parameters.tag_filter != null ? gar.parameters.tag_filter : "") : ""
+      tag_filter_any = gar.parameters != null ? (gar.parameters.tag_filter_any != null ? gar.parameters.tag_filter_any : "") : ""
+      tag_filter_all = gar.parameters != null ? (gar.parameters.tag_filter_all != null ? gar.parameters.tag_filter_all : "") : ""
+      recursive      = false
+      dry_run        = gar.parameters != null ? (gar.parameters.dry_run != null ? gar.parameters.dry_run : false) : false
+      filter         = gar.parameters != null ? "grace-${gar.parameters.grace != null ? gar.parameters.grace : "0"}-keep-${gar.parameters.keep != null ? gar.parameters.keep : "0"}-tag_filter-${gar.parameters.tag_filter != null ? gar.parameters.tag_filter : "no"}-tag_filter_any-${gar.parameters.tag_filter_any != null ? gar.parameters.tag_filter_any : "no"}-tag_filter_any-${gar.parameters.tag_filter_any != null ? gar.parameters.tag_filter_any : "no"}" : "delete-all-untagged-images-recursive"
```

Comment:
typo ?
```suggestion
      filter         = gar.parameters != null ? "grace-${gar.parameters.grace != null ? gar.parameters.grace : "0"}-keep-${gar.parameters.keep != null ? gar.parameters.keep : "0"}-tag_filter-${gar.parameters.tag_filter != null ? gar.parameters.tag_filter : "no"}-tag_filter_any-${gar.parameters.tag_filter_any != null ? gar.parameters.tag_filter_any : "no"}" : "delete-all-untagged-images-recursive"
```

Reply:
Yes it was a typo but I missed `tag_filter_all`. Thanks for findings it! I fixed it.